README.md

```diff
@@ -429,9 +428,27 @@ The returned object is of the form:
 
 Stats will be collected by default, but this can be disabled by setting `opts.stats` to false.
 
-#### `feed.extension(name, message)`
+#### `ext = feed.registerExtension(name, handlers)`
 
-Send an extension message to all connected peers. `name` should be in the list of `extensions` from the constructor. `message` should be a `Buffer`.
+Register a new replication extension. `name` should be the name of your extension and `handlers` should look like this:
+
+```js
+{
+  encoding: 'json' | 'binary' | 'utf-8' | anyAbstractEncoding
+  onmessage (message, peer) {
+    // called when a message is received from a peer
+    // will be decoded using the encoding you provide
+  }
+}
+```
+
+#### `ext.send(message, peer)`
```

Comment:
Is the `peer` param a `Peer` object or is it an ID? Maybe describe that `peer` is the object emitted on the `peer` event.